cmake/kokkos_arch.cmake

```diff
@@ -179,7 +228,7 @@ IF (KOKKOS_ARCH_ARMV81)
 ENDIF()
 
 IF (KOKKOS_ARCH_ARMV8_THUNDERX)
-  SET(KOKKOS_ARCH_ARMV80 ON CACHE BOOL "enable armv80" FORCE)
+  SET(KOKKOS_ARCH_ARMV80 ON) #Not a cache variable
```

Comment:
Why do we leave the cache out?

Reply:
If a variable is not the "origin" of the state, it should not be a cache variable. This follows the CMake best practice of fewest number of cache variables.  A cache variable that causes multiple things to happen should set regular variables so that if it gets changed in the cache, all the dependent regular variables get changed instead of a relic cache variable that conflicts.